src/Avalonia.Base/AvaloniaProperty.cs

```diff
@@ -60,11 +63,12 @@ public abstract class AvaloniaProperty : IEquatable<AvaloniaProperty>, IProperty
             _metadata = new Dictionary<Type, AvaloniaPropertyMetadata>();
 
             Name = name;
-            PropertyType = valueType ?? throw new ArgumentNullException(nameof(valueType));
-            OwnerType = ownerType ?? throw new ArgumentNullException(nameof(ownerType));
+            PropertyType = valueType;
+            OwnerType = ownerType;
             Notifying = notifying;
             Id = s_nextId++;
 
+            metadata.Freeze();
             _metadata.Add(hostType, metadata ?? throw new ArgumentNullException(nameof(metadata)));
```

Comment:
```suggestion
            _metadata.Add(hostType, metadata);
```

Reply:
Thanks! It's already covered by the `ThrowIfNull` at the start of the method, I forgot to remove this `throw`.